### PR TITLE
Add a package for the "repo" multiple Git repository tool

### PR DESCRIPTION
**Pull request by sschuberth**

### CODE REVIEW

File: mingw-w64-git-repo/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,25 @@
+_realname=git-repo
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=0.3.17
+pkgrel=1
+pkgdesc="The multiple Git repository tool"
```
Comment (mingwandroid) [inline mingw-w64-git-repo/PKGBUILD]: Please add (mingw-w64) to end of pkgdesc

Comment (mingwandroid) [inline mingw-w64-git-repo/PKGBUILD]: .. actually, shouldn't repo be in MSYS2-packages as it uses msys2 Python3?

Comment (sschuberth) [inline mingw-w64-git-repo/PKGBUILD]: Hmm, I do want to use this with mingw-python (and with mingw-git from the git-for-windows project). Hence the dependency on `"${MINGW_PACKAGE_PREFIX}-python3"`. What else do I need to do to ensure mingw-python is used?

**Comment by mingwandroid** [comment]
Ah, I didn't look into the details enough, sorry about that. I actually made a PKGBUILD for Google's original version of repo that used msys2 Python3 a good while ago and never added it to the repo.

Problem we have is that:
    #!/usr/bin/env python3

You should change that to:
    #!/mingw64/bin/python3

or
    #!/mingw32/bin/python3

.. so that people without /mingw{32,64}/bin in their PATH (e.g. msys2_shell.bat users) don't have a problem.

**Comment by sschuberth** [comment]
> .. so that people without /mingw{32,64}/bin in their PATH (e.g. msys2_shell.bat users) don't have a problem.

On the other hand, those people (for which `MSYSTEM` is `MSYS`) would not have `/mingw{32,64}/bin` in `PATH` anyway, i.e. they would not be able to (accidentally) run `repo` from there. Or am I missing something?

**Comment by mingwandroid** [comment]
Well, they could explicitly do `/mingw64/bin/repo` which is something I often do when developing MSYS2 PKGBUILDs (usually I run `/mingw64/bin/ag`)

**Comment by mingwandroid** [comment]
Using explicitly the correct Python isn't just better for those without /mingw{32,64}/bin in their PATH, but it also offers protection against some other Python appearing before ours in their PATH. Since MSYS2 is a complete system, we want to ensure that it's our software that gets run and not some other software since we can do without bug reports in those situations.

I hope this sounds reasonable to you?

**Comment by sschuberth** [comment]
> I hope this sounds reasonable to you?

It does. I will make the changes.

**Comment by sschuberth** [comment]
Done. I've now also inlined the patch file to more conveniently switch between `/mingw{32,64}/bin/python3`.

**Comment by mingwandroid** [comment]
Thanks!

**Comment by sschuberth** [comment]
> I actually made a PKGBUILD for Google's original version of repo that used msys2 Python3 a good while ago and never added it to the repo.

Are you now going to add your MSYS2-package, too? :-)

**Comment by mingwandroid** [comment]
Naah, the focus round here is on making native Windows software .. we don't want to step on Cygwin's toes unnecessarily :-)

**Comment by mingwandroid** [comment]
.. I wasn't aware of the Windows-native fork of repo otherwise I'd have done what you did. Btw, don't we need a git dependency for this tool :-) Also, when do you think a merge request for mingw-w64-git will be incoming?

**Comment by sschuberth** [comment]
> Btw, don't we need a git dependency for this tool :-)

Strictly speaking you're right. However, since we don't have git-for-windows' mingw-git package upstream yet I forgot to add it :-)

> Also, when do you think a merge request for mingw-w64-git will be incoming?

I don't know as I've mostly backed down from the git-for-windows project due to dissensions. You would have to ask @dscho for that.

**Comment by dscho** [comment]
> when do you think a merge request for mingw-w64-git will be incoming?

@mingwandroid Real soon now. See [the milestone](https://github.com/git-for-windows/git/milestones/First%20release%20of%20Git%20for%20Windows%20generated%20using%20the%20new%20SDK).

After that, we still have to hash out how to incorporate Git for Windows' changes into the msys2-runtime. I hope to get to that point by mid August.

**Comment by sschuberth** [comment]
@mingwandroid Could we please also get the package binaries for this?

**Comment by mingwandroid** [comment]
@Alexpux builds the binaries and he's on holiday until the end of the month.

**Comment by sschuberth** [comment]
I see, thanks for the info. So let's hope you'll get the [AppVeyor-based package building CI](https://github.com/Alexpux/MSYS2-packages/pull/319) running soon, also for MINGW packages :-)

**Comment by dscho** [comment]
It is not enough to build the mingw-w64-git packages... In order for the unit tests to pass, we need to get the msys2-runtime patches in, too -- in one form or another; I have no problems adjusting my patches as long as Git's test suite still passes afterwards.

This is on my radar, of course: https://github.com/git-for-windows/git/issues/284 As such, I would hate to see things rushed to a point that it is no fun.

**Comment by sschuberth** [comment]
@dscho Neither getting package binaries for mingw-w64-git-repo nor getting the MSYS2 CI to run is rushing anything WRT to mingw-w64-git, so I don't understand what you're talking about.

**Pull request merged**